src/Libraries/RevitNodesUI/GenericClasses.cs

```diff
@@ -0,0 +1,184 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Autodesk.DesignScript.Geometry;
+using Autodesk.DesignScript.Runtime;
+using DSRevitNodesUI;
+using RVT = Autodesk.Revit.DB;
+using RevitServices.Persistence;
+using RevitServices.Transactions;
+
+using Dynamo.Utilities;
+using Dynamo.Models;
+using ProtoCore.AST.AssociativeAST;
+
+namespace DSRevitNodesUI
+{
+
+    /// <summary>
+    /// Generic UI Dropdown node baseclass for Revit Elements.
+    /// This class populates a dropdown with all Revit elements of the specified type.
+    /// It uses a filtered element collector filtering by class.
+    /// </summary>
+    public abstract class CustomRevitElementDropDown : RevitDropDownBase
+    {
+        /// <summary>
+        /// Generic Revit Element Class Dropdown Node
+        /// </summary>
+        /// <param name="name">Name of the Node</param>
+        /// <param name="elementType">Type of Revit Element to display</param>
+        public CustomRevitElementDropDown(string name, Type elementType) : base(name) { this.ElementType = elementType; PopulateItems(); }
+
+        /// <summary>
+        /// Type of Element
+        /// </summary>
+        private Type ElementType
+        {
+            get;
+            set;
+        }
+
+        protected override CoreNodeModels.DSDropDownBase.SelectionState PopulateItemsCore(string currentSelection)
+        {
+            PopulateItems();
+            return SelectionState.Done;
+        }
+        /// <summary>
+        /// Populate the Dropdown menu
+        /// </summary>
+        public void PopulateItems()
```

Comment:
@sharadkjaiswal yes that's true but the PopulateItems of the base class is called before the constructor because its triggered by the base constructor. At this point the EnumerationType is not set yet which I need for populating. That's why I have to call Populate again after setting the EnumerationType.